Neos.ContentRepository.LegacyNodeMigration/Classes/NodeDataToEventsProcessor.php

```diff
@@ -170,11 +170,12 @@ private function resetRuntimeState(): void
 
     private function exportEvent(EventInterface $event): void
     {
+        $normalizedEvent = $this->eventNormalizer->normalize($event);
         $exportedEvent = new ExportedEvent(
-            Uuid::uuid4()->toString(),
-            $this->eventNormalizer->getEventType($event)->value,
-            json_decode($this->eventNormalizer->getEventData($event)->value, true),
-            []
+            $normalizedEvent->id->value,
```

Comment:
i guess this is a functional change as in that we now have stable ids in the export.

Probably a good idea so we have less diff if exporting twice and having the export in version control like the Neos.Demo

Reply:
It's here, isn't it?
https://github.com/neos/neos-development-collection/pull/5016/files#diff-725eb4467b3d93f025036235fdf1935d7aa3140b2a1e767fd7f78409e16a23bcR134

Reply:
Ah i see you're right a bit sneaky i see ^^ (We are sure at that time that its _not_ a decorated event ... thats why it works)

Reply:
Okay in that case i guess its a bit weird that `normalise` is not pure and will return different results on succeeding calls. But the behaviour is not new and required already at other places (we just seem to share the code here now)

So 👍 feel free to merge this ;)

Reply:
Even if it _was_ and even if it would specify the id it would work as long as the id is unique